From 0d0d14e5510127e9c5162e140a9ef6d004c2616a Mon Sep 17 00:00:00 2001
From: Konrad Kujawa <konrad.kujawa@qt.io>
Date: Thu, 22 Jun 2023 15:30:20 +0200
Subject: [PATCH 1/3] Fix include of std::set

With new version of gRPC, compiler complains about missing include of
the std::set. Move include from .cpp to .h files.

Pick-to: 6.6
Change-Id: Ibfef277598cb9e620ab1b693a42564706302c9cc
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
---
 src/tools/qtgrpcgen/qgrpcgenerator.cpp         | 1 -
 src/tools/qtgrpcgen/qgrpcgenerator.h           | 3 ++-
 src/tools/qtprotobufgen/qprotobufgenerator.cpp | 4 ++--
 src/tools/qtprotobufgen/qprotobufgenerator.h   | 2 ++
 4 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/tools/qtgrpcgen/qgrpcgenerator.cpp b/src/tools/qtgrpcgen/qgrpcgenerator.cpp
index 4cab74d..3c2597e 100644
--- a/src/tools/qtgrpcgen/qgrpcgenerator.cpp
+++ b/src/tools/qtgrpcgen/qgrpcgenerator.cpp
@@ -10,7 +10,6 @@
 #include "utils.h"
 #include "options.h"
 
-#include <set>
 #include <google/protobuf/compiler/code_generator.h>
 #include <google/protobuf/stubs/logging.h>
 #include <google/protobuf/stubs/common.h>
diff --git a/src/tools/qtgrpcgen/qgrpcgenerator.h b/src/tools/qtgrpcgen/qgrpcgenerator.h
index 809925e..996d37c 100644
--- a/src/tools/qtgrpcgen/qgrpcgenerator.h
+++ b/src/tools/qtgrpcgen/qgrpcgenerator.h
@@ -4,8 +4,9 @@
 #ifndef QGRPCGENERATOR_H
 #define QGRPCGENERATOR_H
 
-#include <memory>
 #include "generatorbase.h"
+#include <memory>
+#include <set>
 
 namespace google::protobuf {
 class FileDescriptor;
diff --git a/src/tools/qtprotobufgen/qprotobufgenerator.cpp b/src/tools/qtprotobufgen/qprotobufgenerator.cpp
index 5a1ccc2..8eaab81 100644
--- a/src/tools/qtprotobufgen/qprotobufgenerator.cpp
+++ b/src/tools/qtprotobufgen/qprotobufgenerator.cpp
@@ -12,11 +12,10 @@
 #include "utils.h"
 #include "options.h"
 
-#include <cassert>
-
 #include <array>
+#include <cassert>
 #include <numeric>
-#include <set>
+
 #include <google/protobuf/stubs/logging.h>
 #include <google/protobuf/stubs/common.h>
 #include <google/protobuf/io/printer.h>
diff --git a/src/tools/qtprotobufgen/qprotobufgenerator.h b/src/tools/qtprotobufgen/qprotobufgenerator.h
index 8aec33a..6239fd4 100644
--- a/src/tools/qtprotobufgen/qprotobufgenerator.h
+++ b/src/tools/qtprotobufgen/qprotobufgenerator.h
@@ -7,6 +7,8 @@
 
 #include "generatorbase.h"
 
+#include <set>
+
 namespace google::protobuf {
 class FileDescriptor;
 class Descriptor;
